### PR TITLE
Add Arabic locale

### PR DESCRIPTION
**Pull request by tomayac**
Created via Google Translate. Sorry, I don't speak Arabic and would need help from the community to improve the translation.

### CODE REVIEW

**Comment by tomayac** [comment]
deploy-preview-72--frosty-lovelace-4cf67f netlify app_

**Comment by tomayac** [comment]
@shadeed: Thanks for volunteering to take a look! `<3`

File: src/i18n/ar-LB.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  copySVG: 'انسخ ملف SVG',
+  install: 'تثبيت',
+
+  posterizeInputImage: 'صورة الإدخال المتتالية',
+  colorSVG: 'لون SVG',
+  monochromeSVG: 'أحادية اللون SVG',
+
+  colorChannels: 'قنوات اللون',
+  imageSizeAndRotation: 'حجم الإدخال والدوران',
+  imagePreprocessing: 'المدخلات المعالجة المسبقة',
```
Comment (AymenLoukil) [inline src/i18n/ar-LB.js:70]: المعالجة المسبقة

File: src/i18n/ar-LB.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  grayscale: 'تدرج الرمادي',
+  'hue-rotate': 'تدوير هوى',
+  invert: 'عكس',
+  opacity: 'العتامة',
+  saturate: 'التشبع',
+  sepia: 'بني داكن',
+
+  scale: 'حجم',
+  rotation: 'دوران',
+  turdsize: 'قمع البقع',
```
[the remark this answers is not in the document]
Comment (tomayac) [reply src/i18n/ar-LB.js:39]: The translation here is: Suppress speckles (the idea being for an algorithm to be tweaked so its output causes less speckles).

File: src/i18n/ar-LB.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+import languages from './languages.js';
+
+const translations = {
+  red: 'أحمر',
+  green: 'لون أخضر',
```
Comment (tomayac) [inline src/i18n/ar-LB.js:24]:
```suggestion
  green: 'أخضر',
```

File: src/i18n/ar-LB.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+  blue: 'أزرق',
+  alpha: 'ألفا',
+
+  brightness: 'سطوع',
+  contrast: 'مقابلة',
```
Comment (tomayac) [inline src/i18n/ar-LB.js:29]:
```suggestion
  contrast: 'تباين',
```

File: src/i18n/ar-LB.js [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+const translations = {
+  red: 'أحمر',
+  green: 'أخضر',
+  blue: 'أزرق',
+  alpha: 'ألفا',
+
+  brightness: 'سطوع',
+  contrast: 'مقابلة',
+  grayscale: 'تدرج الرمادي',
+  'hue-rotate': 'تدوير هوى',
```
Comment (tomayac) [inline src/i18n/ar-LB.js:31]:
```suggestion
  'hue-rotate': 'تدوير فارق الألوان',
```

File: src/i18n/ar-LB.js [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+  opacity: 'العتامة',
+  saturate: 'التشبع',
+  sepia: 'بني داكن',
+
+  scale: 'حجم',
+  rotation: 'دوران',
+  turdsize: 'قمع البقع',
+  alphamax: 'عتبة الزاوية',
+  minPathSegments: 'الحد الأدنى لطول المسار',
+  strokeWidth: 'عرض السكتة الدماغية',
```
Comment (tomayac) [inline src/i18n/ar-LB.js:42]:
```suggestion
  strokeWidth: 'عرض جرة قلم',
```

File: src/i18n/ar-LB.js [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+  turnpolicy: 'بدوره نهج',
+  opticurve: 'تحسين المنحنيات',
+  opttolerance: 'التسامح الأمثل',
+  showAdvancedControls: 'إظهار خيارات الخبراء',
```
Comment (tomayac) [inline src/i18n/ar-LB.js:46]:
```suggestion
  showAdvancedControls: 'إظهار خيارات متقدمة',
```

File: src/i18n/ar-LB.js [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+  deg: '°',
+  steps: 'خطوات',
+  pixels: 'بكسل',
+  segments: 'شرائح',
+
+  reset: 'إعادة ضبط',
+  resetAll: 'إعادة ضبط الجميع',
+
+  dropFileHere: 'إفلات الملف هنا',
+  openImage: 'صورة مفتوحة',
```
Comment (tomayac) [inline src/i18n/ar-LB.js:58]:
```suggestion
  openImage: 'افتح هذه الصورة',
```

File: src/i18n/ar-LB.js [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+  pasteImage: 'لصق الصورة',
+  copySVG: 'انسخ ملف SVG',
+  install: 'تثبيت',
+
+  posterizeInputImage: 'صورة الإدخال المتتالية',
+  colorSVG: 'لون SVG',
+  monochromeSVG: 'أحادية اللون SVG',
+
+  colorChannels: 'قنوات اللون',
+  imageSizeAndRotation: 'حجم الإدخال والدوران',
```
Comment (tomayac) [inline src/i18n/ar-LB.js:69]:
```suggestion
  imageSizeAndRotation: 'حجم الصورة والدوران',
```

File: src/i18n/ar-LB.js [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+  imagePreprocessing: 'المدخلات المعالجة المسبقة',
```
Comment (tomayac) [inline src/i18n/ar-LB.js:70]:
```suggestion
  imagePreprocessing: 'المعالجة المسبقة',
```

File: src/i18n/ar-LB.js [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+  svgOptions: 'خيارات SVG',
+
+  considerDPR: 'ضع في اعتبارك نسبة وحدات البكسل في الجهاز',
```
Comment (tomayac) [inline src/i18n/ar-LB.js:73]:
```suggestion
  considerDPR: 'اخذ بعين الاعتبار نسبة وحدات البكسل للجهاز',
```

File: src/i18n/ar-LB.js [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+
+  tweak: 'قرص',
```
Comment (tomayac) [inline src/i18n/ar-LB.js:75]:
```suggestion
  tweak: 'تعديل',
```

File: src/i18n/ar-LB.js [hunk 13]
```diff
@@ -0,0 +1,94 @@
...
+  closeOptions: 'قريب',
```
Comment (tomayac) [inline src/i18n/ar-LB.js:76]:
```suggestion
  closeOptions: 'قفل الخيارات',
```

File: src/i18n/ar-LB.js [hunk 14]
```diff
@@ -0,0 +1,94 @@
...
+  saturate: 'التشبع',
+  sepia: 'بني داكن',
+
+  scale: 'حجم',
+  rotation: 'دوران',
+  turdsize: 'قمع البقع',
+  alphamax: 'عتبة الزاوية',
+  minPathSegments: 'الحد الأدنى لطول المسار',
+  strokeWidth: 'عرض جرة قلم',
+  turnpolicy: 'بدوره نهج',
```
Comment (tomayac) [inline src/i18n/ar-LB.js:43]:
```suggestion
  turnpolicy: 'سياسة الدوران',
```

File: src/i18n/ar-LB.js [hunk 15]
```diff
@@ -0,0 +1,94 @@
...
+
+  scale: 'حجم',
+  rotation: 'دوران',
+  turdsize: 'قمع البقع',
+  alphamax: 'عتبة الزاوية',
+  minPathSegments: 'الحد الأدنى لطول المسار',
+  strokeWidth: 'عرض السكتة الدماغية',
+  turnpolicy: 'بدوره نهج',
+  opticurve: 'تحسين المنحنيات',
+  opttolerance: 'التسامح الأمثل',
```
[the remark this answers is not in the document]
Comment (tomayac) [reply src/i18n/ar-LB.js:45]: In English it's "Optimization Tolerance".

File: src/i18n/ar-LB.js [hunk 16]
```diff
@@ -0,0 +1,94 @@
...
+  grayscale: 'تدرج الرمادي',
+  'hue-rotate': 'تدوير فارق الألوان',
+  invert: 'عكس',
+  opacity: 'العتامة',
+  saturate: 'التشبع',
+  sepia: 'بني داكن',
+
+  scale: 'حجم',
+  rotation: 'دوران',
+  turdsize: 'قمع البقع',
```
Comment (tomayac) [inline src/i18n/ar-LB.js:39]:
```suggestion
  turdsize: 'ترطيب الاخاديد',
```

File: src/i18n/ar-LB.js [hunk 17]
```diff
@@ -0,0 +1,94 @@
...
+  alphamax: 'عتبة الزاوية',
+  minPathSegments: 'الحد الأدنى لطول المسار',
+  strokeWidth: 'عرض جرة قلم',
+  turnpolicy: 'سياسة الدوران',
+  opticurve: 'تحسين المنحنيات',
+  opttolerance: 'التسامح الأمثل',
```
Comment (tomayac) [inline src/i18n/ar-LB.js:45]:
```suggestion
  opttolerance: 'درجة التسامح الأمثل',
```

**Comment by tomayac** [comment]
Latest [staging version](https://deploy-preview-72--frosty-lovelace-4cf67f.netlify.app/):

deploy-preview-72--frosty-lovelace-4cf67f netlify app_ (1)

**Comment by shadeed** [comment]
Hey @tomayac, thanks for the invite.

I fixed a few translations and pushed them to the branch `enhance/arabic-lang`.

What does the following mean in the context of the app (In plain words :D)?
- Suppress Speckles
- Corner Threshold
- Turn Policy
- Optimization Tolerance

**Comment by tomayac** [comment]
Hi @shadeed! Thanks for taking the time to contribute! `<3`! I have opened a PR #81  for you.

Here are the answers to your questions, pointing to the [man page](http://potrace.sourceforge.net/potrace.1.html) of Potrace:

- Suppress Speckles: http://potrace.sourceforge.net/potrace.1.html#:~:text=%2Dt%20n%2C%20%2D%2Dturdsize,this%20many%20pixels.
- Corner Threshold: http://potrace.sourceforge.net/potrace.1.html#:~:text=%2Da%20n%2C%20%2D%2Dalphamax,is%20completely%20smooth.
- Turn Policy: http://potrace.sourceforge.net/potrace.1.html#:~:text=%2Dz%20policy%2C%20%2D%2Dturnpolicy,instead%20of%20minority.
- Optimization Tolerance: http://potrace.sourceforge.net/potrace.1.html#:~:text=%2DO%20n%2C%20%2D%2Dopttolerance,expense%20of%20accuracy.

**Pull request merged**